client/block-builder/src/lib.rs

```diff
@@ -153,7 +153,7 @@ where
 			})
 		} else {
 			self.api.map_api_result(|api| {
-				match api.apply_extrinsic_with_context(
+				match api.apply_trusted_extrinsic_with_context(
```

Comment:
We can not really know at this place if the extrinsic is trusted ^^

Reply:
> Have you done any benchmarks?

Haven't been doing any benchmarks for block construction so far, only for block import.
But I think signature verification should not be a bottleneck there (since it is always native?), so do not expect much gain.

Reply:
I thought we always use trusted for block construction from this pr on?

Reply:
The blockbuilder is just a wrapper around the `BlockBuilder` runtime api and you still have the `apply_extrinsic` function. So, either remove `apply_extrinsic` or extend the `BlockBuilder` struct with a function to call this funciton.

Reply:
But there is a compatibility call above to `apply_extrinsic` (or I misunderstand how versioning works)